### PR TITLE
Vlad pubdev 3955

### PR DESCRIPTION
**Pull request by vpatryshev**

### CODE REVIEW

**Comment by vpatryshev** [comment]
Michal, sent it to you, but well, everything's failing everywhere; will check again Monday what's up with all these builds.

File: h2o-core/src/main/java/hex/Model.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public double defaultThreshold() {
       AUTO, Random, Modulo, Stratified
     }
     public enum CategoricalEncodingScheme {
-      AUTO, OneHotInternal, OneHotExplicit, Enum, Binary, Eigen
+      AUTO, OneHotInternal, OneHotExplicit, OneHot, Enum, Binary, Eigen
```
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/Model.java]: We do not need this one for now

File: h2o-core/src/test/java/water/udf/UdfEnumTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    }
+  }
+
+  @Test
+  public void testPlainFrameEncodedViaUdf() throws Exception {
+    EnumColumn x = generate(100);
+    Vec vec = x.vec();
+
+    final EnumColumn enumColumn = new EnumColumn(vec);
+    final UnfoldingFrame<Integer, DataColumn<Integer>> plain = enumColumn.oneHotEncodedFrame("plain");
```
Comment (mmalohlava) [inline h2o-core/src/test/java/water/udf/UdfEnumTest.java]: We have to discuss this - if we are going to promote this one to customers, we need to be sure that the API is stable and usable.

File: h2o-core/src/main/java/hex/Model.java (hunk 1 above)
Comment (mmalohlava) [inline h2o-core/src/main/java/hex/Model.java]: i would propose to remove it for now.

**Comment by mmalohlava** [comment]
@vpatryshev we need one more thing - end-to-end example of using Java API:
load data/apply transformation/build model

We need that for customer.

**Comment by vpatryshev** [comment]
@h2o-ops please test!

File: h2o-core/src/test/java/water/udf/UdfEnumTest.java (hunk 2 above)
Comment (vpatryshev) [inline h2o-core/src/test/java/water/udf/UdfEnumTest.java]: Do you want me to add some tests to validate this statement? Not sure about usability, but stability is measurable.

File: h2o-core/src/main/java/hex/Model.java (hunk 1 above)
Comment (vpatryshev) [inline h2o-core/src/main/java/hex/Model.java]: Will gladly remove it.

**Comment by vpatryshev** [comment]
@h2o-ops please test!

File: h2o-core/src/main/java/water/Scope.java [hunk 3]
```diff
@@ -64,13 +64,18 @@ static void track_internal( Key k ) {
...
-  static public <T extends Keyed> Keyed<T> track_generic(Keyed<T> keyed) {
+  static public <T extends Keyed> T track(T keyed) {
     Scope scope = _scope.get();                   // Pay the price of T.L.S. lookup
     assert scope != null;
     track_impl(scope, keyed._key);
     return keyed;
   }
 
+  @Deprecated
+  static public <T extends Keyed> T track_generic(T keyed) {
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/Scope.java]: This function is only currently used in w2v algo. If you want to rename the function, you don't have to keep the old one.

**Comment by mmalohlava** [comment]
There is something wrong with this PR - contains duplicated changes from master :(

File: h2o-core/src/test/java/water/udf/UdfEnumTest.java (hunk 2 above)
Comment (vpatryshev) [inline h2o-core/src/test/java/water/udf/UdfEnumTest.java]: Can we do something with this?

**Comment by vpatryshev** [comment]
The contents of this branch is now in another

**Pull request closed without merging**